### ISSUE TITLE
TypeError: Cannot set property 'user' of undefined

### ISSUE THREAD

**Issue by paulb-opusvl** [Schlabbi/docker-notify#10]
I'd be grateful if you could advise on this. The images in the container it should be checking are all on hub.docker.com

Docker logs:

```text
Attaching to nextcloud_docker-notify_1
docker-notify_1  | Nov 6, 2020, 18:22:58 UTC:  Checking for updated repositories
docker-notify_1  | Nov 6, 2020, 18:23:00 UTC:  TypeError: Cannot set property 'user' of undefined
docker-notify_1  |     at /usr/src/app/index.js:146:30
docker-notify_1  |     at processTicksAndRejections (node:internal/process/task_queues:93:5)
docker-notify_1  | Nov 6, 2020, 18:23:55 UTC:  Checking for updated repositories
docker-notify_1  | Nov 6, 2020, 18:23:56 UTC:  TypeError: Cannot set property 'user' of undefined
docker-notify_1  |     at /usr/src/app/index.js:146:30
docker-notify_1  |     at processTicksAndRejections (node:internal/process/task_queues:93:5)
docker-notify_1  | Nov 6, 2020, 18:26:37 UTC:  Checking for updated repositories
docker-notify_1  | Nov 6, 2020, 18:36:20 UTC:  Checking for updated repositories
```

This is the container set I'm running. I dropped docker-notify into `docker-compose.override.yml`

```yaml
version: '3.2'

services:
  nextcloud:
    image: ${NEXTCLOUD_IMAGE:-nextcloud}:${NEXTCLOUD_IMAGE_VERSION:-19.0.3}
    volumes:
      - "${PWD}/nextcloud:/var/www/html"
      - "${PWD}/apps:/var/www/html/custom_apps"
      - "${PWD}/config:/var/www/html/config"
      - "${PWD}/data:/var/www/html/data"
    environment: 
      POSTGRES_HOST: db
      POSTGRES_DB: ${NEXTCLOUD_POSTGRES_DB:-nextcloud}
      POSTGRES_PASSWORD: ${NEXTCLOUD_POSTGRES_PASSWORD?REQUIRED}
      POSTGRES_USER: ${NEXTCLOUD_POSTGRES_USER:-nextcloud}      
      NEXTCLOUD_ADMIN_USER: ${NEXTCLOUD_ADMIN_USER:-admin}
      NEXTCLOUD_ADMIN_PASSWORD: ${NEXTCLOUD_ADMIN_PASSWORD?REQUIRED}
      REDIS_HOST: redis
      REDIS_HOST_PORT: 6379
      NEXTCLOUD_TRUSTED_DOMAINS: ${NEXTCLOUD_TRUSTED_DOMAINS}
      SMTP_HOST: ${SMTP_HOST:-mail}
      SMTP_SECURE: ${SMTP_SECURE}
      SMTP_PORT: ${SMTP_PORT:-587}
      SMTP_AUTHTYPE: ${SMTP_AUTHTYPE:-LOGIN}
      SMTP_NAME: ${SMTP_NAME}
      SMTP_PASSWORD: ${SMTP_PASSWORD}
      MAIL_FROM_ADDRESS: ${MAIL_FROM_ADDRESS:-noreply@localhost}
      MAIL_DOMAIN: ${MAIL_DOMAIN}
      OVERWRITEHOST: ${OVERWRITEHOST}
      OVERWRITEPROTOCOL: ${OVERWRITEPROTOCOL:-https}
    depends_on:
      - "db"
    ports:
      - "${PORTBASE}80:80"
    depends_on:
      - db

  redis:
    image: ${REDIS_IMAGE:-redis}:${REDIS_IMAGE_VERSION:-6.0.5-alpine}
    restart: on-failure      

  db:
    environment:
      POSTGRES_DB: ${POSTGRES_DB:-postgres}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD?REQUIRED}
      POSTGRES_USER: ${POSTGRES_USER:-postgres}
      NEXTCLOUD_POSTGRES_DB: ${NEXTCLOUD_POSTGRES_DB:-nextcloud}
      NEXTCLOUD_POSTGRES_USER: ${NEXTCLOUD_POSTGRES_USER:-nextcloud}
      NEXTCLOUD_POSTGRES_PASSWORD: ${NEXTCLOUD_POSTGRES_PASSWORD?REQUIRED}
    image: ${POSTGRES_IMAGE:-postgres}:${POSTGRES_IMAGE_VERSION:-12.2-alpine}
    restart: on-failure
    volumes:
      - "${PWD}/init-db.sh:/docker-entrypoint-initdb.d/init-db.sh"
      - "${PWD}/postgresql:/var/lib/postgresql/data:rw"

```

And this is the `config.json` I'm using:

```json
{
  "notifyServices":[
      {
          "image": "nextcloud:latest",
          "actions": [
              {
                  "type": "mailHook",
                  "instance": "generalMail",
                  "recipient": "info@example.org"
              }
          ]
      }
  ],
  "smtpServer": {
      "generalMail": {
          "host": "mail",
          "port": 1025,
          "username": "docker-notify@example.org",
          "password": "PASSWORD",          
          "sendername": "Docker-Notify",
          "senderadress": "docker-notify@example.org"
      }
  }
}
```

The mail instance is just a mailcatcher so the credentials etc. are unimportant.

**Comment by Schlabbi** [comment]
This looks like an error on my side. The library that I use to talk with docker hub supports pagination. However, I only fetch the first page (pagesize 100) which is the default of this library. In case the tag that you want to look for (the `latest` tag here) is not on the first page, it will throw the error you see there. 

I'll see how I can fix this, because this should not happen.

**Comment by paulb-opusvl** [comment]
Good news. Doesn't sound too bad to sort out then.

**Comment by Schlabbi** [comment]
Should be fixed with c75b5e75f4a16e840221a8b62fad33c603e87d2c. Please let me know if you have any problems. And close this issue if it works now.

**Comment by paulb-opusvl** [comment]
Looks good to me. Many thanks. Look forward to testing this out some more.

```text
docker-notify_1  | Nov 7, 2020, 17:31:56 UTC:  Checking for updated repositories
```
